tpls/desul/include/desul/atomics/Thread_Fence_OpenMP.hpp

```diff
@@ -16,7 +16,8 @@ SPDX-License-Identifier: (BSD-3-Clause)
 namespace desul {
 namespace Impl {
 
-#if _OPENMP > 201800
+// FIXME_OPENMPTARGET_NVHPC: NVHPC compiler does not support omp flush.
+#if _OPENMP > 201800 && !defined(KOKKOS_COMPILER_NVHPC)
```

Comment:
You can't use Kokkos macros in Desul.

Please elaborate, the other branch also has `#pragma omp flush`

Reply:
The issue is with the memory-order-clauses that follow the `omp flush` constructs. They are not supported by the nvhpc compiler.

Reply:
Please make update the comment.  Also use FIXME_OPENMP and/or FIXME_NVHPC